.github/workflows/cmd.yml

```diff
@@ -0,0 +1,417 @@
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `<details><summary>Command help:</summary> 
+            
+            \`\`\`
+            ${{ steps.help.outputs.help }}
+            \`\`\` 
+            
+            </details>`
+            })
+
+      - name: Add confused reaction on failure
+        uses: actions/github-script@v7
+        if: ${{ failure() }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.reactions.createForIssueComment({
+              comment_id: ${{ github.event.comment.id }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              content: 'confused'
+            })
+
+      - name: Add 👍 reaction on success
+        uses: actions/github-script@v7
+        if: ${{ !failure() }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.reactions.createForIssueComment({
+              comment_id: ${{ github.event.comment.id }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              content: '+1'
+            })
+
+  set-image:
+    needs: [ clean, is-org-member ]
+    if: ${{ startsWith(github.event.comment.body, '/cmd') && !contains(github.event.comment.body, '--help') && needs.is-org-member.outputs.member == 'true' }}
+    runs-on: ubuntu-latest
+    outputs:
+      IMAGE: ${{ steps.set-image.outputs.IMAGE }}
+      RUNNER: ${{ steps.set-image.outputs.RUNNER }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - id: set-image
+        run: |
+          BODY=$(echo "${{ github.event.comment.body }}" | xargs)
+          IMAGE_OVERRIDE=$(echo $BODY | grep -oe 'docker.io/paritytech/ci-unified:.*\s' | xargs)
+          
+          cat .github/env >> $GITHUB_OUTPUT
+          
+          if [ -n "$IMAGE_OVERRIDE" ]; then
+              echo "IMAGE=$IMAGE_OVERRIDE" >> $GITHUB_OUTPUT
+          fi
+          
+          if [[ $BODY == "/cmd bench"* ]]; then
+              echo "RUNNER=arc-runners-polkadot-sdk-benchmark" >> $GITHUB_OUTPUT
+          elif [[ $BODY == "/cmd update-ui"* ]]; then
+              echo "RUNNER=arc-runners-polkadot-sdk-beefy" >> $GITHUB_OUTPUT
+          else
+              echo "RUNNER=ubuntu-latest" >> $GITHUB_OUTPUT
+          fi
+
+  cmd:
+    needs: [ set-image ]
+    env:
+      JOB_NAME: 'cmd'
+    runs-on: ${{ needs.set-image.outputs.RUNNER }}
+    container:
+      image: ${{ needs.set-image.outputs.IMAGE }}
+    steps:
+      - name: Get command
+        uses: actions-ecosystem/action-regex-match@v2
+        id: get-pr-comment
+        with:
+          text: ${{ github.event.comment.body }}
+          regex: '^(\/cmd )([-\/\s\w.=:]+)$' # not ";" or && || to reduce risk of injections
```

Comment:
Can you doc what this matches and put some examples please?

How are the commands named? Maybe it should whitelist what can be matched, so that people can also not use " or ' to escape. Is `[\w\-_]+` not enough for the command name?

Reply:
good point, added explanation